File: src/api/address.ts
import api from './config';
import type { ApiResponse } from '../types';

export interface Address {
  id: number;
  userId: number;
  name: string;
  phone: string;
  province: string;
  city: string;
  district: string;
  detailAddress: string;
  isDefault: number;
  createTime?: string;
}

export const getAddressList = (): Promise<ApiResponse<Address[]>> => {
  return api.get('/api/address/list');
};

export const addAddress = (data: {
  name: string;
  phone: string;
  province: string;
  city: string;
  district: string;
  detailAddress: string;
  isDefault?: number;
}): Promise<ApiResponse<Address>> => {
  return api.post('/api/address/add', data);
};

export const updateAddress = (data: {
  id: number;
  name?: string;
  phone?: string;
  province?: string;
  city?: string;
  district?: string;
  detailAddress?: string;
  isDefault?: number;
}): Promise<ApiResponse<Address>> => {
  return api.put('/api/address/update', data);
};

export const deleteAddress = (id: number): Promise<ApiResponse<void>> => {
  return api.delete(`/api/address/delete/${id}`);
};

export const setDefaultAddress = (id: number): Promise<ApiResponse<void>> => {
  return api.put(`/api/address/default/${id}`);
};

File: mall-parent/mall-user/src/main/java/com/mall/user/entity/Address.java
package com.mall.user.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("ums_address")
public class Address {

    @TableId(type = IdType.AUTO)
    private Long id;
    private Long userId;
    private String name;
    private String phone;
    private String province;
    private String city;
    private String district;

    @TableField("`detail`")
    private String detailAddress;

    private Integer isDefault;
    private LocalDateTime createTime;
}
